pitch.md

```diff
@@ -1,69 +1,172 @@
+
+What first interested me about this kind of idea was learning about D. It's essentially
+C++ done right. D offers much closer control over compile-time behavior. In particular,
+D has such features as compile-time "if" statements, compile-time arguments to functions,
+and a much better system than C++'s templates.
+
+The other thing is that I really want to explore type systems.
+First, building a language like this would force me to look at them.
+But also, _having_ a language like this, if it was actually sufficiently expressive,
+would be helpful in learning-through-implementing those ideas.
 
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain? Be careful
-to distinguish between, e.g,. a graphical interface and a linguistic interface._ 
 
+The interface is entirely linguistic. You write some code, and you compile it.
+However, what happens during compilation will also be a part of your code. In some sense,
+errors would happen in "layers" - perhaps not by design, but by reality - in the way that
+there would probably end up being layers to what happens during compilation. (First some mutations,
+then basic type checking, then some kind of verification thing, etc.) So, it would be especially
+important that errors know their source. Specifically, an error that pops up after an AST mutation
+would need to reference the mutation, as well as the original code that was mutated.
+After that, it just needs to be straightforward for people who write compile-time behavior to
+also write error messages.
+
+(You know, that's another thing that might be good. Depending on how much compile-time behavior
+can be coded in, you could have people make variant implementations of things that just had better error
+messages... that's a rare kind of extensibility.)
+
+One interesting point is that, as is often the case for languages with powerful compile-time 
+capabilities, the errors you inevitably get during compilation are not meant just
+to inform you that you've typed something wrong. Rather, they are supposed
+to guide you to what you should do next in your coding. So that may
+also be a (stretch goal) _characteristic_ of the interface.
 
 ### Operation (semantics)
-_What might happen when a program runs? How does a program interact with the
-user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
-might they be communicated to the user?_
 
+A program, as it is running, would probably be a collection of ASTs, or perhaps something
+slightly more general.  Some mutations are run on this collection.
+Then calculations such as type or other property checking are run on the ASTs.
+A final resulting AST, with a lot of annotated information, is produced at the end.
+Theoretically it would either compile to runnable code or transpile to some
+similar-enough language, but it would really be fine if it didn't do this step.
 
```

Comment:
Right, you could end up with a AST but no "run-functionality"
That would still be a major project